### PR TITLE
Add ``Result.all()`` convenience method

### LINKED ISSUE #233: Add all() convenience method
Retrieving the entire set of data from a `Result` object is awkward when using result slices.  To retrieve the full data set for a `Result` using a slice, you must do something like:

``` python
result = Result(...)
data = result[:]
```

...which is a bit awkward.  We should add an `all()` method call that performs the same functionality as `result[:]` so that we could do something like:

``` python
result = Result(...)
data = result.all()
```

### PR DESCRIPTION
**Pull request by smithsz**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant/python-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes
- [x] You have updated the [CHANGES.rst](https://github.com/cloudant/python-cloudant/blob/master/CHANGES.rst) 
- [x] You have completed the PR template below:

## What
Add ``Result.all()`` convenience method.

## How
Performs same functionality as `result[:]`.

## Testing
Includes additional unit test.

## Issues
Fixes #233.

### CODE REVIEW

File: src/cloudant/result.py [hunk 1]
```diff
@@ -359,6 +359,14 @@ def _parse_data(self, data):
         """
         return data.get('rows', [])
 
+    def all(self):
+        """
+        Retrieve the full results data set.
+
+        :return: results data set.
```
Comment (ricellis) [inline src/cloudant/result.py]: Is it really a `set` as in the [built-in](https://docs.python.org/2/library/stdtypes.html#set)? We seem to refer to it as a collection elsewhere.

Comment (smithsz) [inline src/cloudant/result.py]: good spot. no, it's a `list`. I'll update the docstring.

File: src/cloudant/result.py [hunk 2]
```diff
@@ -359,6 +359,17 @@ def _parse_data(self, data):
         """
         return data.get('rows', [])
 
+    def all(self):
+        """
+        Retrieve all results.
+
+        Specifying a ``limit`` parameter in the ``Result`` constructor will
```
Comment (smithsz) [inline src/cloudant/result.py]: The `page_size` parameter is not respected when accessing results via slice. This is already noted in the docs here:
>:param int page_size: Sets the page size for result _iteration_.

Comment (emlaver) [inline src/cloudant/result.py]: @smithsz @ricellis Is it worth pointing to this in the docs or re-stating it?

Comment (smithsz) [inline src/cloudant/result.py]: Done.

**Pull request merged**